finagle-mysql/src/main/scala/com/twitter/finagle/mysql/CanBeParameter.scala

```diff
@@ -0,0 +1,182 @@
+package com.twitter.finagle.exp.mysql
+
+import com.twitter.finagle.exp.mysql.transport.{Buffer, BufferWriter}
+
+trait CanBeParameter[-A] { outer =>
+  /**
+   * Returns the size of the given parameter in its MySQL binary representation.
+   **/
+  def sizeOf(param: A): Int
+
+  /**
+   * Retrieves the MySQL type code for the given parameter.
+   * */
+  def typeCode(param: A): Short
+
+  def write(writer: BufferWriter, param: A): Unit
+}
+
+object CanBeParameter {
+  implicit val stringCanBeParameter = {
+    new CanBeParameter[String] {
+      def sizeOf(param: String): Int = {
+        val bytes = param.getBytes(Charset.defaultCharset)
+        Buffer.sizeOfLen(bytes.size) + bytes.size
```

Comment:
@mosesn Good point. I changed it in many places, but missed this one. (In current Finagle MySQL, `Array#size` is more prevalent than `Array#length`.